plotter/colorbar.go

```diff
@@ -0,0 +1,107 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// ColorBar is a plot.Plotter that draws a color bar legend for a ColorMap.
+type ColorBar struct {
+	// Vertical determines wether the legend will be
+	// plotted vertically or horizontally.
+	// The default is false (horizontal).
+	Vertical bool
+
+	// Colors specifies the number of colors to be
+	// shown in the legend. The default is 255.
+	Colors int
+
+	cm palette.ColorMap
+}
+
+// NewColorBar creates a new ColorBar plotter.
+func NewColorBar(cm palette.ColorMap) *ColorBar {
```

Comment:
I think we could do without a New function if we made `ColorBar` usable with a non-zero `palette.ColorMap`. This would entail defining the behaviour for `Colors == 0`, probably to be one color per point across the varying axis.